### ISSUE TITLE
[Stwo Backend]: Runner

### ISSUE THREAD

**Issue by raphaelDkhn** [gizatechxyz/Luminair#14]
The runner will execute the trace generator for each node in the Luminair graph, returning the results and optionally saving the traces if the user opts to run in proof mode.

Ideally, we should consider implementing a multi-threaded compiler to group independent nodes and execute the graph in parallel when nodes are independent. However, this can be addressed in a separate issue.
